pkgs/by-name/gr/grayjay/package.nix

```diff
@@ -0,0 +1,184 @@
+  autoPatchelfHook,
+  makeDesktopItem,
+  copyDesktopItems,
+  libgcc,
+  krb5,
+  wrapGAppsHook3,
+}:
+let
+  version = "5";
+  src = fetchFromGitLab {
+    domain = "gitlab.futo.org";
+    owner = "videostreaming";
+    repo = "Grayjay.Desktop";
+    tag = version;
+    hash = "sha256-xrbYghNymny6MQrvFn++GaI+kUoOphPQMWcqH47U1Yg=";
+    fetchSubmodules = true;
+    fetchLFS = true;
+  };
+  frontend = buildNpmPackage {
+    name = "grayjay-frontend";
+    inherit version src;
+
+    sourceRoot = "source/Grayjay.Desktop.Web";
+
+    npmBuildScript = "build";
+    npmDepsHash = "sha256-pTEbMSAJwTY6ZRriPWfBFnRHSYufSsD0d+hWGz35xFM=";
+
+    installPhase = ''
+      runHook preInstall
+      cp -r dist/ $out
+      runHook postInstall
+    '';
+  };
+in
+buildDotnetModule {
+  pname = "grayjay";
+
+  inherit version src;
+
+  buildInputs = [
+    openssl
+    libgcc
+    xorg.libX11
+    gtk3
+    glib
+    alsa-lib
+    mesa
+    nspr
+    nss
+    icu
+    krb5
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    wrapGAppsHook3
+    copyDesktopItems
+  ];
+
+  dontWrapGApps = true;
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Grayjay";
+      exec = "Grayjay";
+      icon = "grayjay";
+      comment = "Cross platform media application for streaming and downloading media";
+      desktopName = "Grayjay Desktop";
+      categories = [ "Network" ];
+    })
+  ];
+
+  projectFile = [
+    "Grayjay.ClientServer/Grayjay.ClientServer.csproj"
+    "Grayjay.Engine/Grayjay.Engine/Grayjay.Engine.csproj"
+    "Grayjay.Desktop.CEF/Grayjay.Desktop.CEF.csproj"
+    "FUTO.MDNS/FUTO.MDNS/FUTO.MDNS.csproj"
+    "JustCef/DotCef.csproj"
+  ];
+
+  testProjectFile = [
+    "Grayjay.Desktop.Tests/Grayjay.Desktop.Tests.csproj"
+    "Grayjay.Engine/Grayjay.Engine.Tests/Grayjay.Engine.Tests.csproj"
+  ];
+
+  nugetDeps = ./deps.json;
+
+  dotnet-runtime = dotnetCorePackages.aspnetcore_8_0;
+
+  executables = [ "Grayjay" ];
+
+  preBuild = ''
+    rm -r Grayjay.ClientServer/wwwroot/web
+    cp -r ${frontend} Grayjay.ClientServer/wwwroot/web
+  '';
+
+  postInstall = ''
+    chmod +x $out/lib/grayjay/cef/dotcefnative
+    chmod +x $out/lib/grayjay/ffmpeg
+    rm $out/lib/grayjay/Portable
+    ln -s /tmp/grayjay-launch $out/lib/grayjay/launch
+    ln -s /tmp/grayjay-cef-launch $out/lib/grayjay/cef/launch
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    ln -s $out/lib/grayjay/grayjay.png $out/share/icons/hicolor/scalable/apps/grayjay.png
+  '';
+
+  makeWrapperArgs = [
+    "--chdir"
+    "${placeholder "out"}/lib/grayjay"
+  ];
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  runtimeDeps = [
+    libz
+
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libxcb
+
+    dbus
+    atk
+    cups
+    libdrm
+    expat
+    libxkbcommon
+    pango
+    cairo
+    udev
+    libGL
+    libsecret
+  ];
+
+  passthru.updateScript = gitUpdater { };
```

Comment:
I don't think this update script will properly update the npm hash

Reply:
Yea, this should be replaced with nix-update-script.

Reply:
I've updated this to use nix-update-script but I'm not 100% the updating is working correctly. The time it takes to build is slowing down my ability to test things, but I'll try to keep testing.